pkg/controller/support/credentials/cloudauth/acr/acr.go

```diff
@@ -79,34 +93,39 @@ func newProvider(ctx context.Context, logger logr.Logger) (*acrProvider, error)
 	}
 
 	return &acrProvider{
-		logger:                logger.WithName("acr-auth-provider"),
 		tenantID:              settings.Values[auth.TenantID],
 		servicePrincipalToken: token,
 	}, nil
 }
 
-func (a *acrProvider) authenticate(ctx context.Context, server string) (*types.AuthConfig, error) {
+func (a *acrProvider) authenticate(ctx context.Context, logger logr.Logger, server string) (*types.AuthConfig, error) {
+	logger = logger.WithName("acr-auth-provider")
+
 	match := acrRegex.FindAllString(server, -1)
 	if len(match) != 1 {
-		return nil, fmt.Errorf("invalid ACR url: %q should match %v", server, acrRegex)
+		err := fmt.Errorf("ACR URL is invalid: %q should match pattern %v", server, acrRegex)
+		logger.Info(err.Error())
+		return nil, err
 	}
 
 	loginServer := match[0]
-	err := retry(ctx, a.logger, 3, func() error {
+	err := retry(ctx, logger, 3, func() error {
 		return a.servicePrincipalToken.EnsureFreshWithContext(ctx)
 	})
 	if err != nil {
+		logger.Error(err, "Failed to refresh AAD token.")
 		return nil, fmt.Errorf("failed to refresh AAD token: %w", err)
```

Comment:
- If we use `logger.Error`, the users will get a stacktrace in the build logs, which I don't think we want
- I would also DRY this up and not use `"."` in the logs

```go
err = fmt.Errorf("AAD token refresh failure: %w", err)
logger.Info(err.Error())

return nil, err
```